### DISCUSSION TITLE
Trying to 'push' a value via flask

### DISCUSSION THREAD

**Discussion by vukmirovic98** [JoelBender/bacpypes#402] [Q&A]
I am trying to 'push' a value via flask in a GET request so I could access it via node-red, and would actually like to edit some values in a POST request, would be great if you had the time to review my code and point me on what I am doing wrong. Thanks

Code: https://pastebin.com/naHWjJ4G

**Comment by JoelBender** [comment]
This bit of code:
```python
test_av.statusFlags = str(args['statusflags']) 
```
It run in the context of `p2` while the BACpypes stack is running in `p1`.  The easiest way to so this is to package up the attribute change and put it in the queue that the BACpypes `run()` function will see by calling `deferred()`, and make sure that the status flags value is a list of ones and zeros or bit names:
```python
deferred(setattr, test_av, 'statusFlags', StatusFlags(args['statusflags']))
```

**Reply by vukmirovic98** [reply]
Could you write me a demo file?
Thanks

**Comment by JoelBender** [comment]
Ugh!  This is much harder than I expected.  With flask and BACpypes running in separate processes, it much harder to exchange content between them, using [shared memory](https://docs.python.org/3/library/multiprocessing.shared_memory.html) would require serializing the object and some kind of handshake for knowing when the shared version of the object is current.  Perhaps if flask was running in the main thread and you had BACpypes running in a separate thread it would be manageable, but I don't know how that would work with misty and I don't have any way of testing it.

I attempted to build this using just IPv4 and not MSTP and running flask with debug=True makes it fail because it attempts to build the stack twice and the port binding fails.

So the call to `deferred()` that I suggested works for threads but not multiple processes.

**Comment by ChristianTremblay** [comment]
BAC0 have something that works but far from optimal.

Have a look to the repo.
